AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -836,7 +845,7 @@ - (void)transitionLayoutWithSizeRange:(ASSizeRange)constrainedSize
 {
   // Passed constrainedSize is the the same as the node's current constrained size it's a noop
   ASDisplayNodeAssertMainThread();
-  if ([self shouldCalculateLayoutWithConstrainedSize:constrainedSize parentSize:constrainedSize.max] == NO) {
+  if (_calculatedDisplayNodeLayout->isValidForConstrainedSizeParentSize(constrainedSize, constrainedSize.max)) {
```

Comment:
Does this case ever happen?  We call setNeedsLayout above in the method that calls this one.  Shouldn't this method also call setNeedsLayout, and never have this bail check?

It seems common to me to have a layout transition where the constrainedSize would be the same, but we need to recalculate the layout...  cc @nguyenhuy for thoughts too.